include/rocksdb/options.h

```diff
@@ -390,6 +390,18 @@ struct DBOptions {
   // Default: true
   bool paranoid_checks = true;
 
+  // If true, track WALs in MANIFEST and verify them on recovery.
+  //
+  // If a WAL is tracked in MANIFEST but is missing from disk on recovery,
+  // or the size of the tracked WAL is larger than the WAL's on-disk size,
+  // an error is reported and recovery is aborted.
+  //
+  // If a WAL is not tracked in MANIFEST, then no verification will happen
+  // during recovery.
+  //
+  // Default: false
+  bool track_and_verify_wals_in_manifest = false;
```

Comment:
If this option is not currently connected to anything (we don't have diff stacks unfortunately), please drop in something like

    // FIXME(cheng): This option is part of a work in progress and does not yet work

in case through some mixup or unexpected absence from work, a release only has some of your changes